javascript/challenges/queuesWithStacks/queue-with-stacks.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const {Stack} = require('../../data-structures/stacksAndQueues/stacks-and-queues.js');
+
+class PseudoQueue {
+
+  constructor () {
+    this.stack1 = new Stack();
+    this.stack2 = new Stack();
+    this.length = 0;
+  }
+
+  enqueue(value){
+    this.stack1.push(value);
+    this.length++;
+  }
+
+
+  dequeue(){
+
+    if(this.stack1.isEmpty()){
+      throw new RangeError('Cannot dequeue from empty stack.');
+    } else {
+      while(!this.stack1.isEmpty()){
+        let poppedItem = this.stack1.pop();
+        this.length--;
+        this.stack2.push(poppedItem);
+      }
+    }
+
+    const valOfPopped = this.stack2.pop();
+
+    while(!this.stack2.isEmpty()){
+      let itemPopped = this.stack2.pop();
+      this.stack1.push(itemPopped);
+      this.length++;
+    }
+
```

Comment:
Something you could do for less code would be to check if stack1 is empty or not, throw the error that it is. Then push into stack2, popping off of stack1 (hope that makes sense) and then pop off stack2. Because when you push from stack1 to stack2 it will reverse the order!